h2o-extensions/xgboost/src/test/java/hex/tree/xgboost/XGBoostUtilsTest.java

```diff
@@ -46,6 +69,116 @@ public void parseFeatureScores() throws IOException, ParseException {
     }
   }
 
+  @Test
+  public void testCSRPredictionComparison_cars() {
+    //Cars is a 100% dense dataset (useful edge case)
+    try {
+      Scope.enter();
+      final String response = "cylinders";
+      final Frame frame = TestUtil.parse_test_file("smalldata/junit/cars.csv");
+      Scope.track(frame);
+      final Frame testFrame = TestUtil.parse_test_file("smalldata/testng/cars_test.csv");
+      Scope.track(testFrame);
+
+      testCSRPredictions(frame, response, testFrame);
+    } finally {
+      Scope.exit();
+    }
+  }
+
+  @Test
+  public void testCSRPredictionComparison_airlines() {
+    try {
+      Scope.enter();
+      final String response = "IsDepDelayed";
+
+      final Frame frame = TestUtil.parse_test_file("smalldata/testng/airlines.csv");
+      Scope.track(frame);
+      final Frame testFrame = TestUtil.parse_test_file("smalldata/testng/airlines_test.csv");
+      Scope.track(testFrame);
+      
+      testCSRPredictions(frame, response, testFrame );
+    } finally {
+      Scope.exit();
+    }
+  }
+
+  @Test
+  public void testCSRPredictionComparison_airQuality() {
+    try {
+      Scope.enter();
+      final String response = "Ozone";
+
+      final Frame frame = TestUtil.parse_test_file("smalldata/testng/airquality_train1.csv");
+      Scope.track(frame);
+      final Frame testFrame = TestUtil.parse_test_file("smalldata/testng/airquality_validation1.csv");
+      Scope.track(testFrame);
+
+      testCSRPredictions(frame, response, testFrame );
+    } finally {
+      Scope.exit();
+    }
+  }
+
+  @Test
+  public void testCSRPredictionComparison_prostate() {
+    try {
+      Scope.enter();
+      final String response = "GLEASON";
+
+      final Frame frame = TestUtil.parse_test_file("smalldata/testng/prostate_train.csv");
+      Scope.track(frame);
+      final Frame testFrame = TestUtil.parse_test_file("smalldata/testng/prostate_test.csv");
+      Scope.track(testFrame);
+
+      testCSRPredictions(frame, response, testFrame );
+    } finally {
+      Scope.exit();
+    }
+  }
+
+  @Test
+  public void testCSRPredictionComparison_synthetic_sparse() {
```

Comment:
This test will currently fail, predictions of dense and sparse are not the same.